guardian/grid
# T1628 usage rights summary

## What does this change?

The aim of this chage is to provide a framework for adapting the information presented in the 'Rights and restrictions' section of the metadata panel when an image (or multupe images) are selected. The actual text and icon displayed is controlled by a separate component - gr-usagerights-bbc.tsx - this controls the range of usage rights subject to the additional information - the logic dictating each set of rights and the icon and text to be displayed.

For example;

![Screenshot 2024-04-09 at 09 51 18](https://github.com/guardian/grid/assets/128470622/e178073d-cbd2-4d4c-92fc-387872bbb3f1)

In the case of multiple images selected it would look like;

<img width="957" alt="Screenshot 2024-04-09 at 10 11 32" src="https://github.com/guardian/grid/assets/128470622/5ccd8837-b77b-4cc8-ab72-ff3e5132f879">

For usage rights not included in the list of options the display is effectively unchanged;

![Screenshot 2024-04-09 at 09 52 31](https://github.com/guardian/grid/assets/128470622/385181bb-a271-4825-80fd-0a0518ac2fd6)

## How should a reviewer test this change?

Ensure that when an image matches one fo the criteria in collection in the usage rights tsx file that the usage rights section displays the correct additional text and icon and that when the usage rights of the image is not in that collection that the display remains unchanged.

Ensure that when multiple images are selected the 'info panel' displays usage rights correctly and consistently with current behaviour.

Access to this functionality is controlled by a config variable in the Kahuna config - set usageRightsSummary = true to gain access to functionality (setting to false turns it off) and if the parameeter is not present it defaults to false.

## Tested? Documented?
